python/kserve/kserve/triton/triton_configuration.py

```diff
@@ -0,0 +1,538 @@
+# Copyright 2024 The KServe Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+from argparse import ArgumentParser
+from typing import List, Dict, Any
+
+import tritonserver
+from tritonserver import (
+    InstanceGroupKind,
+    ModelControlMode,
+    RateLimitMode,
+    RateLimiterResource,
+)
+from tritonserver._api._server import ModelLoadDeviceLimit
+
+
+class TritonOptions:
+    @staticmethod
+    def add_cli_args(parser: ArgumentParser) -> ArgumentParser:
+        triton_arg_group = parser.add_argument_group(
+            "triton server options", "Arguments for Triton server"
+        )
+        triton_arg_group.add_argument(
```

Comment:
Not sure how we can keep this in sync, does this list exist in triton which we can import here like we do for vllm ?

Reply:
No. There is no argument parser in the lib.